examples/research_projects/resadapter/README.md

```diff
@@ -0,0 +1,86 @@
+
+**Paper**: https://arxiv.org/abs/2403.02084
+**Code**: https://github.com/bytedance/res-adapter
+**Project Page**: https://res-adapter.github.io/
+
+### Training
+
+```bash
+python3 train_sd_resadapter.py \
+  --pretrained_model_name_or_path <MODEL> \
+  --dataset_name <DATASET> \
+  --dataset_config_name <DATASET_SPLIT> \
+  --image_column image \
+  --caption_column prompt \
+  --validation_prompt "beautiful face, youthful appearance, ultra focus, face iluminated, face detailed, ultra focus, dreamlike images, pixel perfect precision, ultra realistic;Award-winning photo of a mystical fox girl fox in a serene forest clearing, sunlight" \
+  --validation_prompt_sep ";" \
+  --num_validation_images 5 \
+  --validation_epochs 1 \
+  --validation_heights 256 384 768 768 1024 \
+  --validation_widths 256 832 768 1280 1024 \
+  --validation_inference_steps 20 \
+  --output_dir sd-resadapter \
+  --cache_dir . \
+  --seed 42 \
+  --nearest_resolution_multiple 64 \
+  --max_resolution 1920 \
+  --random_flip \
+  --train_batch_size 4 \
+  --num_train_epochs 5 \
+  --gradient_accumulation_steps 1 \
+  --learning_rate 1e-5 \
+  --adam_beta1 0.95 \
+  --adam_beta2 0.99 \
+  --checkpointing_steps 500 \
+  --rank 8 \
+  --report_to wandb
+```
+
+### Inference
+
+The following code assumes the checkpoints were generated using the training script. When the training script completes, you should see LoRA weights and state dict for UNet GroupNorm layers in your model/output directory.
+
+```python
+import torch
+from diffusers.pipelines.pipeline_utils import DiffusionPipeline
+
+model_id = "<MODEL_ID>"
+resadapter_dir = "<LORA_WEIGHTS_DIRECTORY>"
+pipe = DiffusionPipeline.from_pretrained(
+    model_id,
+    variant="fp16",
+    torch_dtype=torch.float16,
+).to("cuda")
+
+# load resadapter
+unet_groupnorm_state_dict = torch.load(f"{resadapter_dir}/groupnorm_state_dict.pt")
```

Comment:
We shouldn't serialize using torch pickle here. `safetensors` is a required dependency of the library so let's use that here. 

Also, can we try to load `f"{resadapter_dir}/groupnorm_state_dict.pt"` from the Hub instead? Something like:


```py
from huggingface_hub import hf_hub_download
import safetensors.torch

state_dict_path = hf_hub_download(repo_id="repo_id", filename="groupnorm_state_dict.pt")
unet_groupnorm_state_dict = safetensors.torch.load_file(state_dict_path)
```

Does this make sense?